### ISSUE TITLE
mx.mod.Module.predict error

### ISSUE THREAD

**Issue by qinhui99** [dmlc/mxnet#5660]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:

Compiler:

Package used (Python/R/Scala/Julia): python

MXNet version: 0.93,0.94

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution: 2.7

If you are using R package, please provide

R `sessionInfo()`:

## Error Message: ./matrix_op-inl.h:712: Check failed: param.begin[i] < shp[i] && param.end[i] <= shp[i] && param.begin[i] < param.end[i] 
Please paste the full error message, including stack trace.

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.
[test_module_iris.txt](https://github.com/dmlc/mxnet/files/887541/test_module_iris.txt)

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.mod = mx.mod.Module(symbol=net,
                    context=mx.cpu(),
                    data_names=['data'],
                    label_names=['softmax_label'])

logging.basicConfig(level=logging.INFO)

epoch_num=15
mod.fit(train_iter,
        eval_data=val_iter,
        optimizer='adam',
        optimizer_params={'learning_rate':.1},
        eval_metric='acc',
        num_epoch=epoch_num)

print('Finished training')

y = mod.predict(val_iter,num_batch=1)
2.new_samples = np.array(
    [[6.4, 3.2, 4.5, 1.5], [6.0, 2.2, 5.0, 1.5], [6.9, 3.2, 5.7, 2.3], [5.6, 2.8, 4.9, 2.0], [7.7, 2.8, 6.7, 2.0],
     [6.7, 3.3, 5.7, 2.1], [7.2, 3.2, 6.0, 1.8], [6.2, 2.8, 4.8, 1.8], [6.4, 2.8, 5.6, 2.1], [6.1, 3.0, 4.9, 1.8]],
    dtype=float)

result=mod.predict(mx.io.NDArrayIter(data=new_samples,label=None,batch_size=1))
3.

## What have you tried to solve it?

1. I try to use the mx.model.FeedForward.predict(), it worked. But it is a deprecated api. :( 
2.
3.

**Comment by chaonan99** [comment]
I encounter this problem today and it seems that you can not use the same module on data iter in a new shape. To solve this, you can build a new module, bind its shape with the new data_iter and set parameter to the trained module. Check this example:

https://github.com/chaonan99/mxnet-play/blob/master/deeplearningbook/xor/config/xor.py#L87

**Comment by qinhui99** [comment]
Thanks. It works.
